index.js

```diff
@@ -534,7 +564,14 @@
   function Number$prototype$equals(other) {
     return typeof this === 'object' ?
       equals(this.valueOf(), other.valueOf()) :
-      isNaN(this) && isNaN(other) || this === other && 1 / this === 1 / other;
+      isNaN(this) && isNaN(other) || this === other;
+  }
+
+  //  Number$prototype$lte :: Number ~> Number -> Boolean
+  function Number$prototype$lte(other) {
+    return typeof this === 'object' ?
+      lte(this.valueOf(), other.valueOf()) :
+      isNaN(other) && isNaN(this) || this <= other;
```

Comment:
We could replace this with `this < other || equals(this, other)`. It reads nicely and avoids repeating the `isNaN` checks, though it would be less efficient. What do you think, Gabe?

If we decide to stick with the current approach, let's switch the `isNaN` checks to match `Number$prototype$equals`.